fence/blueprints/data/indexd.py

```diff
@@ -845,6 +858,28 @@ def _generate_anonymous_google_storage_signed_url(
         )
         return final_url
 
+    def bucket_name(self):
+        resource_path = self.get_resource_path()
+
+        bucket_name = None
+        try:
+            bucket_name = resource_path.split("/")[0]
+        except Exception as exc:
+            logger.error("Unable to get bucket name from resource path. {}".format(exc))
+
+        return bucket_name
+
+    def file_name(self):
+        resource_path = self.get_resource_path()
+
+        file_name = None
+        try:
+            file_name = resource_path.split("/")[1]
```

Comment:
I actually think this needs to be 
```suggestion
            file_name = resource_path.split("/")[-1]
```

B/c this won't work if the file is within a couple folders like: "gs://bucket-name/folder1/folder2/file.txt"